components/catalogs/back/src/Application/Handler/GetMappedProductsHandler.php

```diff
@@ -128,18 +127,21 @@ private function getProductMappingSchema(string $catalogId): array
     }
 
     /**
-     * @param Product $product
+     * @param RawProduct $product
      */
     private function getProductAttributeValue(array $product, string $attributeCode, ?string $locale, ?string $scope): string
     {
-        if (\array_key_exists($attributeCode, $product['values'])) {
-            /** @var ProductValue $attributeValues */
-            foreach ($product['values'][$attributeCode] as $attributeValues) {
-                if ($attributeValues['locale'] === $locale && $attributeValues['scope'] === $scope) {
-                    return (string) $attributeValues['data'];
-                }
-            }
+        $value = '';
+        $scope ??= '<all_channels>';
+        $locale ??= '<all_locales>';
+
+        if (\array_key_exists($attributeCode, $product['raw_values']) &&
+            \array_key_exists($scope, $product['raw_values'][$attributeCode]) &&
+            \array_key_exists($locale, $product['raw_values'][$attributeCode][$scope])
+        ) {
+            $value = $product['raw_values'][$attributeCode][$scope][$locale];
         }
-        return '';
+
+        return $value;
```

Comment:
```suggestion
        return $product['raw_values'][$attributeCode][$scope][$locale] ?? $value;
```

I think you remove all calls to array_key_exists